src/DatePicker/DatePicker.js

```diff
@@ -28,6 +29,63 @@ class DatePicker extends Component {
         this.popoverRef = React.createRef();
     }
 
+    /**
+     * Function tries to format any date string into the format specified by
+     * props.dateFormat (or ISO_DATE_FORMAT as fallback).
+     * It will try to use multiple similar formats, defined in saneDateLookup, to
+     * create a Moment.js date object from given dateString.
+     * It will return an empty string if Moment.js is unable to create a valid
+     * date object from given string (for e.g. text only strings).
+     * @param  {String|Object} date to format.
+     * @returns {String} formatted date string.
+     */
+    getFormattedDateStr = (date) => {
+        if (date) {
+            const momentDateObj = this.getMomentDateObj(date);
+            if (momentDateObj && momentDateObj.isValid()) {
+                const { dateFormat } = this.props;
```

Comment:
If the user provides a locale but not a dateFormat, I would think that the locale's should be used. Right now because there's a default set (ISO), it would always override the locale (e.g. look at the localized datepickers on the docs site which are normally DD/MM/YYYY).

Reply:
Thanks, that's correct.